polaris-react/src/components/Button/Button.module.scss

```diff
@@ -97,6 +97,11 @@
   color: var(--pc-button-color_hover);
   outline: var(--p-border-width-050) solid var(--p-color-border-focus);
   outline-offset: var(--p-space-025);
+
+  // Disable focus-ring mixin to prevent overlap of focus-ring and outline
+  &::after {
+    content: none;
+  }
```

Comment:
Disabling the usage of the focus-ring mixin in TooltipOverlay.module.scss would fix the overlap of focus styles specifically for Tooltip that's rendering a Button, but for all other cases the Tooltip would no longer have focus styles. Opted for this approach and also did not import styles from common to use the `no-focus-ring` mixin because it seemed intentional to no longer import common styles in the larger Button refactor PR.